File: php/features.php
<?php
declare(strict_types=1);

// XkcdComics SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class XkcdComicsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new XkcdComicsBaseFeature();
            case "test":
                return new XkcdComicsTestFeature();
            default:
                return new XkcdComicsBaseFeature();
        }
    }
}

File: py/features.py
# XkcdComics SDK feature factory

from feature.base_feature import XkcdComicsBaseFeature
from feature.test_feature import XkcdComicsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: XkcdComicsBaseFeature(),
        "test": lambda: XkcdComicsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
